basic_bootloader/src/bootloader/mod.rs

```diff
@@ -163,22 +165,46 @@ impl<'a> SafeUsizeWritable for TxDataBufferWriter<'a> {
     }
 }
 
-impl<S: EthereumLikeTypes> BasicBootloader<S> {
+// TODO: type of Metadata is hardcoded for now, will be cleaned in future PRs
+impl<S: EthereumLikeTypes<Metadata = zk_ee::system::metadata::Metadata<EthereumIOTypesConfig>>>
+    BasicBootloader<S>
+{
     /// Runs the transactions that it loads from the oracle.
     /// This code runs both in sequencer (then it uses ForwardOracle - that stores data in local variables)
     /// and in prover (where oracle uses CRS registers to communicate).
     pub fn run_prepared<Config: BasicBootloaderExecutionConfig>(
-        oracle: <S::IO as IOSubsystemExt>::IOOracle,
+        mut oracle: <S::IO as IOSubsystemExt>::IOOracle,
         result_keeper: &mut impl ResultKeeperExt,
         tracer: &mut impl Tracer<S>,
     ) -> Result<<S::IO as IOSubsystemExt>::FinalData, BootloaderSubsystemError>
     where
         S::IO: IOSubsystemExt,
     {
         cycle_marker::start!("run_prepared");
+
+        // TODO: this will be moved to metadata_op in a future PR
+        let metadata: S::Metadata = {
+            use zk_ee::metadata_markers::basic_metadata::BasicBlockMetadata;
+            use zk_ee::oracle::query_ids::BLOCK_METADATA_QUERY_ID;
+            use zk_ee::oracle::IOOracle;
+            use zk_ee::system::metadata::Metadata;
+            let block_level_metadata: BlockMetadataFromOracle =
+                oracle.query_with_empty_input(BLOCK_METADATA_QUERY_ID)?;
+
+            let metadata = Metadata {
+                tx_origin: Default::default(),
+                tx_gas_price: Default::default(),
+                block_level_metadata,
+            };
+
+            if metadata.block_gas_limit() > MAX_BLOCK_GAS_LIMIT {
```

Comment:
Should we add a similar check for tx gas limit?

Reply:
Added a check